config/boards/orangepi5.conf

```diff
@@ -35,17 +43,65 @@ function post_family_config_branch_current__orangepi5_use_mainline_uboot() {
 	}
 
 	function write_uboot_platform_mtd() {
-		flashcp -v -p "$1/u-boot-rockchip-spi.bin" /dev/mtd0
+		FILES=$(find "$1" -maxdepth 1 -type f -name "u-boot-rockchip-spi*.bin")
+		if [ -z "$FILES" ]; then
+			echo "No SPI image found."
+			exit 1
+		fi
+
+		MENU_ITEMS=()
+		i=1
+
+		# Read the files into an array
+		while IFS= read -r file; do
+			filename=$(basename "$file")
+			MENU_ITEMS+=("$i" "$filename" "")
+			((i++))
+		done <<< "$FILES"
+
+		# If there is only one image, we can skip the dialog
+		if [[ $i -eq 2 ]]; then
+			flashcp -v -p "$1/${MENU_ITEMS[1]}" /dev/mtd0
+			return
+		fi
+
+		[[ -f /etc/armbian-release ]] && source /etc/armbian-release
+		backtitle="Armbian for $BOARD_NAME install script, https://www.armbian.com"
+
+		CHOICE=$(dialog --no-collapse \
+			--title "armbian-install" \
+			--backtitle $backtitle \
+			--radiolist "Choose SPI image:" 0 56 4 \
+			"${MENU_ITEMS[@]}" \
+			3>&1 1>&2 2>&3)
+
+		if [ $? -eq 0 ]; then
+			flashcp -v -p "$1/${MENU_ITEMS[($CHOICE*3)-2]}" /dev/mtd0
+		else
+			echo "No SPI image chosen."
+			exit 1
+		fi
 	}
 }
 
+function pre_config_uboot_target__orangepi5_patch_uboot_bootconfig_hack_for_sata() {
```

Comment:
Almost OK, except orangepi-5-sata-rk3588s_defconfig is not exist in `/usr/lib/u-boot` directory.

Reply:
@rpardini do you have an idea how i can handle this one better?